src/misc/HashMap.ts

```diff
@@ -9,92 +9,114 @@ import { EqualsFunction, HashFunction } from "./HashSet.js";
 
 interface Entry<Key extends IComparable, Value> { key: Key, value: Value; }
 
-export class HashMap<Key extends IComparable, Value> {
-    private data: { [key: string]: Array<Entry<Key, Value>>; };
+export class HashMap<TKey extends IComparable, TValue> {
+
+    /**
+     * Threshold for using hashing amd searching the bucket instead of a linear search.
+     * Set to 0 to disable linear search and always use the hash function.
+     */
+    public static LINEAR_SEARCH_THRESHOLD = 5;
+
+    private _values: TValue[] = [];
+    private _keys: TKey[] = [];
```

Comment:
As these are especially marked with a leading underscore (which my linter rules disallow btw.) I'd say they should be declared as real private (e.g. `#values`). Maybe even make all private fields really private. I come from a C++ background where private really means private :-)

Reply:
I agree 100%! I was a bit worried about the synthetic sugar that tsc adds when using private fields would have to much impact on performance. And as `values` and `keys` are both functions I decided to go with the classic `_` prefix.

I'll do a quick test if real `#` private fields have a noticeable performance impact and if not make the changes suggested 👍

Reply:
Btw. the release bundle is not build by TSC. That creates only the typings. `esbuild` is what creates the release and I hope it directly takes over private members.

Reply:
Oh yes, your right! I forgot that `esbuild` doesn't use `tsc`. 

And yes your right, it takes over the private members as is. No strange hidden map and helper functions that you sometimes get with tsc.

Reply:
Ok changes applied - I left the `hashFunction` and `equalsFunction` as design-time privates.